docs/building-apps/extracting-wrapper-flags.rst

```diff
@@ -0,0 +1,103 @@
+Extracting flags from the wrapper compilers
+===========================================
+
+If you cannot use the wrapper compilers for some reason, there are
+multiple supported ways to extract the compiler/linker flags that you
+will need.
+
+Using the ``--showme`` option
+-----------------------------
+
+The wrapper compilers all support a ``--showme`` command line option
+that will show what commands would have been invoked.
+
+.. code-block:: sh
+
+   # Show the flags necessary to compile MPI C applications
+   shell$ mpicc --showme:compile
+
+   # Show the flags necessary to link MPI C applications
+   shell$ mpicc --showme:link
+
+   # Show all the flags necessary to build MPI C applications
+   shell$ mpicc --showme
+
+.. note:: If you pass ``--showme`` *and additional command line
+   parameters* to the wrapper compiler, sure to *also* pass in a
+   filename.  Otherwise, the ``--showme`` functionality will not
+   display output as expected.
+
+.. warning:: It is almost never a good idea to hard-code these results
+   in a ``Makefile`` (or other build system).  It is almost always
+   best to run (for example) ``mpicc --showme:compile`` in a dynamic
+   fashion to find out what you need.  For example, GNU Make allows
+   running commands and assigning their results to variables:
+
+   .. code-block:: make
+
+      MPI_COMPILE_FLAGS = $(shell mpicc --showme:compile)
+      MPI_LINK_FLAGS = $(shell mpicc --showme:link)
+
+      my_app: my_app.c
+              $(CC) $(MPI_COMPILE_FLAGS) my_app.c $(MPI_LINK_FLAGS) -o my_app
+
+Using ``pkg-config``
+--------------------
+
+Alternatively, Open MPI also installs ``pkg-config(1)`` configuration
+files under ``$libdir/pkgconfig``.  If ``pkg-config`` is configured to
+find these files (e.g., if you add ``$libdir/pkgconfig`` |mdash| which
+is usually ``$prefix/lib/pkgconfig`` |mdash| to the
+``PKG_CONFIG_PATH`` environment variable), then compiling / linking
+Open MPI programs can be performed like this:
+
+.. code-block:: sh
+
+   shell$ export PKG_CONFIG_PATH=/opt/openmpi/lib/pkgconfig
+   shell$ gcc hello_world_mpi.c -o hello_world_mpi -g \
+               `pkg-config ompi-c --cflags --libs`
+   shell$
+
+Open MPI supplies multiple ``pkg-config`` configuration files; one for
+each different wrapper compiler (language):
+
+* ``ompi``: Synonym for ``ompi-c``; Open MPI applications using the C
+  MPI bindings
+* ``ompi-c``: Open MPI applications using the C MPI bindings
+* ``ompi-cxx``: Open MPI applications using the C MPI bindings
+* ``ompi-fort``: Open MPI applications using the Fortran MPI bindings
+
```

Comment:
We need to add a note about dependencies here.  We don't handle them correctly (we'll add -lpmix to libs.private, rather than depending on the PMIx pkg-config module.